src/Data/Array/Polarized/Pull.hs

```diff
@@ -1,21 +1,85 @@
 {-# LANGUAGE LinearTypes #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 
--- | This module is intended to be imported qualified, e.g.
+-- | This module provides pull arrays.
+--
+-- Use pull arrays to force fusion (i.e., force GHC to not allocate arrays in
+-- the garbage collector memory) in a computation that only uses lists
+-- as /intermediate/ results without operations that \"write\" to a list.
+-- You should think about pull arrays as arrays you can \"pull\" or read from,
+-- zip with and map over. These are arrays that work nicely as arguments to
+-- functions and poorly as arrays to write to.
+--
+-- Pull arrays fit within a larger framework for controlling when
+-- garbage collector memory is allocated; please see "Data.Array.Polarized".
+--
+-- Import this module qualified for clarity and to avoid
+-- name clashes.
```

Comment:
What's wrong with the prosaic, and less patronising, “This module is intented to be imported qualified.”?

Reply:
Oh nothing -- Beth Aitman (in her talk about documentation) recommended directness and interestingly there was some discussion in the audience about whether imperatives like "do this" v.s. "this is a good idea" sounded impolite. It seemed to vary across cultures with the British and Asian at the most polite end and the Americans at the other. I'll lean towards a more European style going forward. (To my culturally-American ears "Import this module like this ..." sounds completely normal and not-patronizing.)

Reply:
Actually, I don't think I am opposed to “Import this module qualified”. But the “for clarity and to avoid name clashes” sounds a bit petty and redundant, I don't know. Like assuming that the reader doesn't know what qualifying modules is for.

Or “Prefer importing this module qualified”. All this is fine.

As a side note, whether the intent for a module is to be imported qualified or unqualified will affect the design of its function names. This is the reason for the “intended to be imported qualified” phrasing that I had.